docs/topic_guides/comparison_templates.ipynb

```diff
@@ -0,0 +1,323 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Out-of-the-box Comparisons for specific data types\n",
+    "\n",
+    "Similarity is defined differently for types of data (e.g. names, dates of birth, postcodes, addresses, ids). The [Comparison Template Library](customising_comparisons.ipynb#method-2-using-the-comparisontemplatelibrary) contains functions to generate read-made comparisons for a variety of data types.\n",
+    "\n",
+    "Below are examples of how to structure comparisons for a variety of data types."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Date Comparisons\n",
+    "\n",
+    "Date comparisons are generally structured as: \n",
+    "\n",
+    "- Null level  \n",
+    "- Exact match  \n",
+    "- Fuzzy match ([using metric of choice](comparators.md))  \n",
+    "- Interval match (within X days/months/years)  \n",
+    "- Else level\n",
+    "\n",
+    "The [comparison_template_library](#method-2-using-the-comparisontemplatelibrary) contains the `date_comparison` function which gives this structure, with some pre-defined parameters, out-of-the-box."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Comparison 'Exact match vs. Dates within levenshtein thresholds 1, 2 vs. Dates within the following thresholds Year(s): 1, Year(s): 10 vs. anything else' of \"date_of_birth\".\n",
+      "Similarity is assessed using the following ComparisonLevels:\n",
+      "    - 'Null' with SQL rule: \"date_of_birth_l\" IS NULL OR \"date_of_birth_r\" IS NULL\n",
+      "    - 'Exact match' with SQL rule: \"date_of_birth_l\" = \"date_of_birth_r\"\n",
+      "    - 'Levenshtein <= 1' with SQL rule: levenshtein(\"date_of_birth_l\", \"date_of_birth_r\") <= 1\n",
+      "    - 'Levenshtein <= 2' with SQL rule: levenshtein(\"date_of_birth_l\", \"date_of_birth_r\") <= 2\n",
+      "    - 'Within 1 year' with SQL rule: \n",
+      "        abs(date_diff('year', \"date_of_birth_l\", \"date_of_birth_r\")) <= 1\n",
+      "    \n",
+      "    - 'Within 10 years' with SQL rule: \n",
+      "        abs(date_diff('year', \"date_of_birth_l\", \"date_of_birth_r\")) <= 10\n",
+      "    \n",
+      "    - 'All other comparisons' with SQL rule: ELSE\n",
+      "\n"
```

Comment:
You can ignore this if you disagree, but perhaps a directory tree with some written explanations for each level would go nicely here?

It would just act as a compliment for the "human readable" version, which is still difficult to interpret if you're seeing it for the first time.

Reply:
See [here](https://github.com/moj-analytical-services/splink/blob/master/docs/topic_guides/customising_comparisons.ipynb) for an example.

Reply:
Yea you are probably right - I will throw one together

Reply:
Have included a directory tree for both functions. I think it is clearer - thanks for the prompt!